kernel/src/drivers/ps2/io.rs

```diff
@@ -130,3 +53,78 @@ pub enum DeviceCommand {
     SetScancode = 0xF0,
     Reset = 0xFF,
 }
+
+/// Represents an error returned by PS/2
+#[derive(Copy, Clone, Debug)]
+pub enum Ps2Error {
+    NoData,
+    DeviceUnavailable,
+}
+
+/// Sends a controller command without a return
+pub fn command(cmd: ControllerCommand) -> Result<(), Ps2Error> {
```

Comment:
(FTR: it's not stale ;P) IMO this solution is a bit weak though on a type system level -- you could abuse it by passing a returning command or a command with data to that function.

Reply:
You couldn't. `ControllerCommand` and `ControllerReturnCommand` are distinct enums.

Reply:
(See edit)

Reply:
Oh, it didn't go through :/

Reply:
Fixed.